timeboost/src/lib.rs

```diff
@@ -58,6 +67,16 @@ impl Timeboost {
             None
         };
 
+        let provider = PubSubProvider::new(cfg.chain_config.parent.ws_url.clone()).await?;
+        let start_ts = cfg
+            .prev_committee
+            .as_ref()
+            .map(|c| c.effective_timestamp())
+            .unwrap_or_default();
+        let events =
+            CommitteeInfo::new_committee_stream(&provider, start_ts, &cfg.chain_config.parent)
+                .await?;
```

Comment:
IIUC this would go over all committees from the beginning if `prev_committee` is `None`. We should probably include the effective timestamp of the current committee in the `TimeboostConfig` as this would be the starting point for events to consider.

If possible it would be even better if the `KeyManager` contract can include the block number in its committee information, then we would not need to search for blocks, but know exactly where to start.